core/wait/wait.go

```diff
@@ -12,72 +12,77 @@ import (
 
 var RetryHttpErrorStatusCodes = []int{http.StatusBadGateway, http.StatusGatewayTimeout}
 
-type WaitFn func() (res interface{}, done bool, err error)
-
-type Handler struct {
-	fn                WaitFn
+// AsyncActionCheck reports whether a specific async action has finished.
+//   - waitFinished == true if the async action is finished, false otherwise.
+//   - response contains data regarding the current state of the resource targeted by the async action (if applicable). resource != nil if waitFinished == true.
+//   - err != nil if there was an error checking if the async action finished, or if it finished unsuccessfully.
+type AsyncActionCheck[T any] func() (waitFinished bool, response *T, err error)
```

Comment:
What was the reasoning behind switching the output order?

Reply:
Just that `waitFinished` is "more important", in the sense that this function gets called repeatedly until is equals `true`. Also `WaitWithContext`'s signature matches this one without the `waitFinished`, and it looked weird for it to be between the other two outputs